src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqScenarioRepository.kt

```diff
@@ -23,14 +26,38 @@ class JooqScenarioRepository(dsl: DSLContext) : JooqRepository(dsl), ScenarioRep
     override fun getScenarios(descriptionIds: Iterable<String>): List<Scenario>
     {
         return dsl.select(SCENARIO_DESCRIPTION.fieldsAsList())
-                .select(SCENARIO.TOUCHSTONE)
+                .select(SCENARIO.TOUCHSTONE, COVERAGE_SET.ACTIVITY_TYPE)
                 .fromJoinPath(SCENARIO_DESCRIPTION, SCENARIO)
+                .leftJoin(COVERAGE_SET)
+                .on(SCENARIO.FOCAL_COVERAGE_SET.eq(COVERAGE_SET.ID))
                 .where(SCENARIO_DESCRIPTION.ID.`in`(descriptionIds.distinct().toList()))
                 .fetch()
                 .groupBy { it[SCENARIO_DESCRIPTION.ID] }
-                .map { mapScenario(it.value) }
+                .map { mapScenarioWithFocalActivityType(it.value) }
+                .sortedBy { it.activityType }
+                .map { it.scenario }
+
+    }
+
+    private fun mapScenarioWithFocalActivityType(input: List<Record>): ScenarioAndFocalActivityType
+    {
+        val first = input.first()
+        val scenario = mapScenario(input)
+        val activityType = first[COVERAGE_SET.ACTIVITY_TYPE]
+        val activityTypeEnum = if (activityType != null)
+        {
+            mapper.mapEnum<ActivityType>(activityType)
+        }
+        else
+        {
+            null
+        }
```

Comment:
One small readability suggestion: We could add a method `mapNullableEnum` method to mapper, that handles this null check for us. Then this code could be the more readable: 
```
val activityType = mapper.mapNullableEnum(first[COVERAGE_SET.ACTIVITY_TYPE])
```

Reply:
Nice idea